### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by mwcvitkovic**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -7,3 +7,6 @@ dependencies:
   - pytorch=1.6.0
   - torchvision=0.7.0=py37_cpu
   - tqdm=4.47.0
+  - pip
```
Comment (mwcvitkovic) [inline environment.yml:10]: Make sure to add any new dependencies you install to `environment.yml`.  This shows how to add things you install via pip.

File: models/resnet_official.py [hunk 2]
```diff
@@ -41,23 +46,23 @@ def __init__(
             groups: int = 1,
             base_width: int = 16,
             dilation: int = 1,
-            #norm_layer: Optional[Callable[..., nn.Module]] = None
-            norm = None
+            # norm_layer: Optional[Callable[..., nn.Module]] = None
```
Comment (mwcvitkovic) [inline models/resnet_official.py:49]: All the changes to the files in `models/` are just stylistic.  E.g. having a space after the `#` in comments.  Feel free to ignore.  The traditional standard for formatting python code is called [PEP8](https://www.python.org/dev/peps/pep-0008/).

File: train_supervised.py [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
-import os
-import replicate
+
 from models import wide_resnet, resnet_official
 
 dir_path = os.path.dirname(os.path.realpath(__file__))
 p = os.path.join(dir_path, 'params.json')
 with open(p, 'r') as f:
     params = json.load(f)
-    f.close()
```
Comment (mwcvitkovic) [inline train_supervised.py:17]: The `with` statement means you don't have to close the file here.  Once the code is out of the `with` block, python closes it for you.  

Technically what's going on here is that the file object returned from `open()` is a [Context Manager](https://docs.python.org/3/reference/datamodel.html#context-managers).  (No need to read and understand the docs at that link, just if you're curious.)   Context Managers are a way of defining a safe little runtime for code that might fail.

File: train_supervised.py [hunk 4]
```diff
@@ -30,6 +31,7 @@
 # normalize all pixel values to be in [-1, 1] and add Gaussian noise with mean zero, variance gaussian_noise_var
 transform = transforms.Compose(
     [transforms.ToTensor(),
+     # [0.4914, 0.4822, 0.4465], [0.2488, 0.2453, 0.2633]
```
Comment (mwcvitkovic) [inline train_supervised.py:34]: Not sure where the (0.5, 0.5, 0.5) mean and std dev came from, but the values in the comments here are the empirical mean and std computed over the training data (per channel).  That's technically what you should be using in Normalize, though it won't make much of a difference at all to performance.

File: train_supervised.py [hunk 5]
```diff
@@ -47,36 +49,35 @@
...
-#model = wide_resnet.Wide_ResNet(depth=depth, widen_factor=widen_factor, dropout_rate=dropout_rate, num_classes=num_classes)
+# model = wide_resnet.Wide_ResNet(depth=depth, widen_factor=widen_factor, dropout_rate=dropout_rate, num_classes=num_classes)
 model = resnet_official.wrn_28_2().to(device)
 
-#define the optimizer and criterion
+# define the optimizer and criterion
 criterion = nn.CrossEntropyLoss()
-optimizer = optim.Adam(model.parameters(), lr=learning_rate, betas=[.9, .999])
-
+optimizer = optim.Adam(model.parameters(), lr=learning_rate, betas=(.9, .999))
```
Comment (mwcvitkovic) [inline train_supervised.py:61]: Unless you specifically need some functionality of a python `list`, like `append`ing or `extend`ing, always use tuples as containers in python.  Immutability is highly preferred to avoid headaches.

File: train_supervised.py [hunk 6]
```diff
@@ -47,36 +49,35 @@
...
-#create a real training loop
+# create a real training loop
 for epoch in range(num_epochs):
-    for i, data in tqdm(enumerate(trainloader), total=len(trainset) // train_batch_size + 1):
+    for i, (inputs, labels) in tqdm(enumerate(trainloader), total=len(trainset) // train_batch_size + 1):
 
-        #obtain data
-        inputs, labels = data[0].to(device), data[1].to(device)
+        # obtain data
+        inputs, labels = inputs.to(device), labels.to(device)
```
Comment (mwcvitkovic) [inline train_supervised.py:76]: You can unpack the `data` tuple right in the for loop, which is more "Pythonic".

File: train_supervised.py [hunk 7]
```diff
@@ -85,28 +86,27 @@
...
-            #check accuracy on validation set
-            for data in testloader:
-                images, labels = data[0].to(device), data[1].to(device)
-                logits = model(images)
-                loss = nn.CrossEntropyLoss(reduction='none')(logits, labels).cpu().numpy()
+            # check accuracy on validation set
+            for test_inputs, test_labels in testloader:
+                test_inputs, test_labels = test_inputs.to(device), test_labels.to(device)
+                logits = model(test_inputs)
+                loss = nn.CrossEntropyLoss(reduction='none')(logits, test_labels).cpu().numpy()
```
Comment (mwcvitkovic) [inline train_supervised.py:100]: Always best to use new variable names (i.e. `test_labels` instead of reusing `labels`) instead of overloading variable names.

Sometimes it's handy, like when performing a series of related operations (for example, look at lines 70 through 85 in `resnet_official` - that's a fine use of overloading).  But in this case, I'd worry about accidentally using train labels when test labels were intended, or vice versa.
